### PR TITLE
[Bug][flink-runtime][connectors-v2] Flink register table Environment  The running mode is set to`job.mode`

### PR DESCRIPTION
**Pull request by zhilinli123**
issues: #4809

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by zhilinli123** [comment]
PTAL:@TyrantLucifer @hailin0 @ashulin  thks

**Comment by zhilinli123** [comment]
```
hive Source:

create table  test3 (
  name string,
  age int,
  emails ARRAY<STRING>,
  phones map<STRING, string>
);
INSERT INTO TABLE test3 
VALUES ('list',12,array("zhangsan@qq.com", "lisi@sohu.com", "wangwu@sina.com"), map('home', "13987167117", 'office', "15178968888"));

hive Sink:

create table  test3_sink (
  name string,
  age int,
  emails ARRAY<STRING>,
  phones map<STRING, string>
);

Conf :
env {
  # You can set spark configuration here
  execution.parallelism = 1
  job.mode = "BATCH"
}



source {
  Hive {
    table_name = "default.test3"
    metastore_uri = "thrift://localhost:9083"
    result_table_name = "user"
  }
}


transform {
  # If you would like to get more information about how to configure seatunnel and see full list of transform plugins,
  # please go to https://seatunnel.apache.org/docs/category/transform
  Replace {
    source_table_name = "user"
    result_table_name = "user_1"
    replace_field = "name"
    pattern = "s"
    replacement = "w"
    is_regex = true
    replace_first=true
  }
}

sink {
# Console{ source_table_name = "user" }
Hive {
    source_table_name = "user_1"
    table_name = "default.test3_sink"
    metastore_uri = "thrift://localhost:9083"
 }
}


table result test3_sink :
liwt    12      ["zhangsan@qq.com","lisi@sohu.com","wangwu@sina.com"]   {"office":"15178968888","home":"13987167117"}

```
### Class:org.apache.seatunnel.example.flink.v2.SeaTunnelApiExample

image

**Comment by zhilinli123** [comment]
image
Rerun CI?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/split/FileSourceSplitEnumerator.java [hunk 1]
```diff
@@ -123,8 +123,10 @@ public int currentUnassignedSplitSize() {
 
     @Override
     public void registerReader(int subtaskId) {
+        Set<Integer> readers = context.registeredReaders();
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/split/FileSourceSplitEnumerator.java]: Why change it?

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/split/FileSourceSplitEnumerator.java (hunk 1 above)
Comment (zhilinli123) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/split/FileSourceSplitEnumerator.java]: > Why change it?

I just reconfirmed that there is no such problem,  I have restored it

**Comment by zhilinli123** [comment]
PTAL: @TyrantLucifer @hailin0 Thank you for your help in reviewing.

File: seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/SourceExecuteProcessor.java [hunk 2]
```diff
@@ -71,13 +71,17 @@ public List<DataStream<Row>> execute(List<DataStream<Row>> upstreamDataStreams)
             } else {
                 sourceFunction = new SeaTunnelParallelSource(internalSource);
             }
+            boolean bounded =
+                    internalSource.getBoundedness()
+                                    == org.apache.seatunnel.api.source.Boundedness.BOUNDED
+                            ? true
+                            : false;
```
Comment (TaoZex) [inline seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/SourceExecuteProcessor.java]:
```suggestion
            boolean bounded = internalSource.getBoundedness() == org.apache.seatunnel.api.source.Boundedness.BOUNDED
```

Comment (zhilinli123) [reply seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/SourceExecuteProcessor.java]: Done

**Review by TaoZex — APPROVED** [review]
LGTM

**Comment by zhilinli123** [comment]
image
 Ci rerun

**Comment by zhilinli123** [comment]
PTAL: @TyrantLucifer @TaoZex

**Comment by ic4y** [comment]
Waiting for the e2e issue to be fixed.

**Comment by zhilinli123** [comment]
> Waiting for the e2e issue to be fixed.

Done

**Comment by zhilinli123** [comment]
ci run @ic4y @liugddx

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Review by ic4y — APPROVED** [review]

**Comment by ic4y** [comment]
@TyrantLucifer PTAL

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**